src/hooks/custom-hooks/useOAuth.ts

```diff
@@ -36,13 +36,14 @@ const useOAuth = (): UseOAuthReturn => {
     const { OAuth2Logout: oAuthLogout } = useOAuth2(oAuthGrowthbookConfig, WSLogoutAndRedirect);
 
     const onRenderAuthCheck = useCallback(() => {
-        if (
-            (!isEndpointPage && !isAuthorized && !isAuthorizing) ||
-            (!isEndpointPage && error?.code === 'InvalidToken')
-        ) {
-            oAuthLogout();
+        if (!isEndpointPage) {
+            if (error?.code === 'InvalidToken') {
+                oAuthLogout();
+            } else if (!isAuthorized && !isAuthorizing) {
+                window.open(oauthUrl, '_self');
```

Comment:
ah hold on, can you check if this works in firefox, i think it might not work in that browser, if it doesnt work, we can instead use window.location.href = oauthUrl